packages/@aws-cdk/aws-stepfunctions/test/state-machine.test.ts

```diff
@@ -315,6 +315,71 @@ describe('State Machine', () => {
     });
   });
 
+  test('Instantiate a State Machine with a task assuming a literal roleArn from same account', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    // WHEN
+    const role = iam.Role.fromRoleName(stack, 'Role', 'example-role');
```

Comment:
Yeah this is what I want to test, but with a role in a different account.

```ts
const app = new cdk.App();
const stack = new cdk.Stack(app, 'Stack1', {
  env: { account: '111111111111' },
});
const stack2 = new cdk.Stack(app, 'Stack2', {
  env: { account: '222222222222' },
});

const role = iam.Role.fromRoleName(stack2, 'Role', 'example-role');
new sfn.StateMachine(stack, 'MyStateMachine', {...});

Template.fromStack(stack).hasResourceProperties('AWS::StepFunctions::StateMachine', {
  DefinitionString: {
    '{"Credentials":{"RoleArn":"arn:aws:iam::111111111111:role/example-role"}}'
  },
});
```

Reply:
I've added a similar test in https://github.com/aws/aws-cdk/pull/23012/commits/93303a395a159915460b39d90b94833da13ba49f

Reply:
ah whoops I was only looking at one of the commits!